### PR TITLE
Update docs: add configuration, remove 'upbound'.

### PR DESCRIPTION
**Pull request by AlainRoy**
### Description of your changes
* Add documentation on the "configuration" group
* Add documentation on the new install-completions command
* Remove documentation on the "upbound" group since it was removed.
* Tweaked documentation on the CLI configuration to distinguish it from the new configurations. 

Note: The only alpha command that was documented was the upbound group, so the entire alpha section was removed. 

I have:

- [X] Read and followed Upbound's [contribution process](https://git.io/fj2m9).

### How has this code been tested
With the markdown viewer in VS Code.

### CODE REVIEW

File: docs/commands.md [hunk 1]
```diff
@@ -159,6 +158,45 @@ Format: `up controlplane kubeconfig <cmd> ...` Alias: `up ctp kubeconfig
       connect to the specified control plane. This kubeconfig file will be
       configured to use the current cluster as the control plane.
 
+## Configurations
+Format: `up configuration <cmd> ...` Alias: `up cfg <cmd> ...`
+
+Commands in the **Configuratione** group are used to manage and interact with
+configurations.
```
Comment (jastang) [inline docs/commands.md:165]:
```suggestion
control plane configurations. A control plane configuration is a Crossplane
Configuration that satisfies the [xpkg specification] and stores its package
contents in a GitHub repository that the control plane is aware of.
```

I think your hunch will be right about some initial confusion about the overloaded term so suggesting to explain it further. Corresponding suggestion for named link below

File: docs/commands.md [hunk 2]
```diff
@@ -471,77 +509,18 @@ server.
...
-  `upbound-system`): Kubernetes namespace used for installing and managing
-  Upbound.
+Once the completion commands have been installed, you can use the
+tab key to auto-complete up commands.
 
 <!-- Named Links -->
 [Upbound Software License]: https://licenses.upbound.io/upbound-software-license.html
 [Upbound Marketplace]: https://www.upbound.io/registry
-[configuration documentation]: configuration.md
+[up configuration documentation]: configuration.md
```
Comment (jastang) [inline docs/commands.md:526]:
```suggestion
[up configuration documentation]: configuration.md
[xpkg specification]: https://docs.crossplane.io/v1.10/reference/xpkg
```

File: docs/commands.md [hunk 3]
```diff
@@ -159,6 +158,45 @@ Format: `up controlplane kubeconfig <cmd> ...` Alias: `up ctp kubeconfig
...
+      or installed the Upbound GitHub app, your web browser will be opened to do so.
+- `list`
+    - Behavior: Lists all configuration.
+- `get <configuration name>`
+    - Behavior: Gets a single configuration
+- `delete <configuration name>`
+    - Behavior: Deletes the specified configuration
+- `template list`
+    - Behavior: List the configuration templates you can specify as a `template-id`
+      to the create command.
```
Comment (jastang) [inline docs/commands.md:183]: Not sure if we'll support more interaction with `template` but should we put it into a subgroup for consistency?

File: docs/commands.md [hunk 4]
```diff
@@ -471,77 +509,18 @@ server.
         - `--verbose = BOOL`: Run server with verbose logging.
     - Behavior: Runs the Crossplane language server.
 
-## Alpha
+## Install Shell Completions
```
Comment (jastang) [inline docs/commands.md:512]: Should this nest under ##Top-Level for consistency?

**Review by jastang — COMMENTED** [review]
Thanks @AlainRoy! Did a first pass and left some suggestions.

I'd also suggest adding a note in `maturity.md` to indicate that `up controplane` will be moved to stable maturity level in the upcoming release, though the examples are still useful:

```
> Note: `up controlplane` and all subcommands have been moved to stable maturity
level starting from `v0.16.0`. `up alpha ctp` commands will continue to work for
backwards compatibility with systems utilizing it such as CI.
```

File: docs/commands.md [hunk 5]
```diff
@@ -14,7 +14,6 @@ Groups:
 - [UXP](#uxp)
 - [XPKG](#xpkg)
 - [XPLS](#xpls)
-- [Alpha](#upbound)
```
[the remark this answers is not in the document]
Comment (AlainRoy) [reply docs/commands.md:17]: `up xpkgs xp-extract` was not documented in the Alpha section. I'm open to separately documenting it there, but I think it should be a separate PR.

**Comment by AlainRoy** [comment]
> Thanks @AlainRoy! Did a first pass and left some suggestions.
> 
> I'd also suggest adding a note in `maturity.md` to indicate that `up controplane` will be moved to stable maturity level in the upcoming release, though the examples are still useful:
> 
> ```
> > Note: `up controlplane` and all subcommands have been moved to stable maturity
> level starting from `v0.16.0`. `up alpha ctp` commands will continue to work for
> backwards compatibility with systems utilizing it such as CI.
> ```

@jastang -- I'm going to leave the maturity.md alone for now and focus on the primary documentation. Let's talk about that document separately. I'm wondering if we want to do something totally different. Whether we do or not, I think it can be a separate PR.

File: docs/commands.md [hunk 6]
```diff
@@ -159,6 +165,47 @@ Format: `up controlplane kubeconfig <cmd> ...` Alias: `up ctp kubeconfig
...
+control plane configurations. A control plane configuration is a Crossplane
+Configuration that satisfies the [xpkg specification] and stores its package
+contents in a GitHub repository that the control plane is aware of.
+
+- `create <configuration name>`
+    - Flags:
+        - `--template-id = STRING`: (Required) Name of the configuration template
+        to use.
+        - `--context = STRING`: (Required) Name of the GitHub account or org
+           to use. The configuration template will be clonde into this Github org.
```
Comment (jastang) [inline docs/commands.md:181]:
```suggestion
           to use. The configuration template will be cloned into this Github org.
```

File: docs/commands.md (hunk 3 above)
Comment (jastang) [reply docs/commands.md:192]: Sounds good to me!

**Pull request merged**